### PR TITLE
Use OpenStack models for waiting for conditions

### PR DESCRIPTION
**Pull request by carpnick**
See #117 for original issue.  Thinking about it more, what I think we need to do is use the models from Fog a bit more.  DIdnt like #119 originally since it is still arbitrary and have to wait the whole time.  Included  the model logic in both server create and volume create here.  This is a WIP, but wanted to get feedback.

### CODE REVIEW

File: lib/kitchen/driver/openstack_version.rb [hunk 1]
```diff
@@ -22,6 +22,6 @@ module Kitchen
   #
   # @author Jonathan Hartman <j@p4nt5.com>
   module Driver
-    OPENSTACK_VERSION = '2.1.0'
+    OPENSTACK_VERSION = '2.1.1'
```
Comment (jjasghar) [inline lib/kitchen/driver/openstack_version.rb]: Please remove this, i'll take care of the versioning.

**Comment by jjasghar** [comment]
All in all, this seems more robust, so i'm all of for it. I'd like to run it through some testing, but I like where this is going.

File: lib/kitchen/driver/openstack_version.rb (hunk 1 above)
Comment (carpnick) [inline lib/kitchen/driver/openstack_version.rb]: No worries on this.   This was for local testing reasons.  Wanted to know I was installing my change.  Will revert in final PR.

**Comment by carpnick** [comment]
Take a look @jjasghar when you get a moment. Thanks.

**Comment by carpnick** [comment]
Its worth mentioning that I excluded the spec folder from code coverage.  So I had to add quite a bit more unit tests to  get the number back up to 90% since obviously all of the spec lines were executed via rspec.  So overall this might show as percentage going down, but I would argue, the total lines covered is dramatically higher after excluding the spec folder.

**Comment by carpnick** [comment]
Resynced after latest master changes.

**Comment by jjasghar** [comment]
Awesome, thanks for the update. It looks like Coveralls is failing now, can you update the tests to reflect a positive change?

**Comment by carpnick** [comment]
Please see comment earlier.  I excluded the spec folder from coverage due to my thought of it being invalid coverage additions.  Coverage usually, you only want for code being shipped.  Once I excluded the spec folder, I had to add a bunch of unit tests to get it back to where it is currently (since spec folder was artificially keeping the number higher)

So 3 options here 
  * I go and add even more unit tests for stuff that was never covered
  * I allow the spec folder in the coverage report 
  *  accept as is.

Thoughts?

**Comment by jjasghar** [comment]
Ah yeah sorry about that.

Yeah if you can add more tests for code coverage i'm good with that. I know we don't have 100% so moving towards that is always a plus.

So, yeah option :one: :metal:

**Comment by carpnick** [comment]
I think this is scope creep for this type of PR(new-feature).  The code coverage number this PR is being compared against is no longer equal.  

But I will go back through and add some UT...

**Comment by jjasghar** [comment]
Yeah that's valid. It's your call, i'm good with this in general; i just hate seeing the :x: in the tests.

Granted I haven't actually validated it myself; i was planning on when I've finished my travels.

**Comment by carpnick** [comment]
Let me see what I can do

**Comment by carpnick** [comment]
@jjasghar -Better :)

Missed 3 lines of code that we just need to mock out really.  But overall I think we are there now.

**Comment by jjasghar** [comment]
:+1: This worked as expected. Awesome. I'll give it week before merging to ideally let someone else test it too.

**Comment by carpnick** [comment]
@jjasghar  anything blocking this?

**Comment by jjasghar** [comment]
I'll release a `--pre` here today, then give it a little more time.

**Comment by carpnick** [comment]
closes #117

**Pull request merged**